pkg/portfwd/client.go

```diff
@@ -52,74 +47,43 @@ func HandleUDPConnection(ctx context.Context, client *guestagentclient.GuestAgen
 		return
 	}
 
-	g, _ := errgroup.WithContext(ctx)
-
-	g.Go(func() error {
-		buf := make([]byte, 65507)
-		for {
-			n, addr, err := conn.ReadFrom(buf)
-			if err != nil {
-				// https://pkg.go.dev/net#PacketConn does not mention io.EOF semantics.
-				if errors.Is(err, io.EOF) {
-					return nil
-				}
-				return err
-			}
```

Comment:
This was wrong in current code, we ignored the last n byte read if we got io.EOF. Fixed in #2971.

Reply:
Seems that it has the same issue:

```go
func (proxy *UDPProxy) Run() {
	readBuf := make([]byte, UDPBufSize)
	for {
		read, from, err := proxy.listener.ReadFrom(readBuf)
		if err != nil {
			// NOTE: Apparently ReadFrom doesn't return
			// ECONNREFUSED like Read do (see comment in
			// UDPProxy.replyLoop)
			if !isClosedError(err) {
				log.Debugf("Stopping udp proxy (%s)", err)
			}
			break
		}
```

It ignores the last n byte read.

It is calling listener.ReadFrom() which seems to have the same issue:

```go
func (pk *pseudoLoopbackPacketConn) ReadFrom(bytes []byte) (n int, addr net.Addr, err error) {
	n, remoteAddr, err := pk.PacketConn.ReadFrom(bytes)
	if err != nil {
		return 0, nil, err
	}

	remoteAddrIP, _, err := net.SplitHostPort(remoteAddr.String())
	if err != nil {
		return 0, nil, err
	}
	if !IsLoopback(remoteAddrIP) {
		return 0, nil, fmt.Errorf("pseudoloopback forwarder: rejecting non-loopback remoteAddr %q", remoteAddr)
	}
	return n, remoteAddr, nil
}
```

Reply:
We can raise PR with gvisor-tap-vsock to fix the UDPProxy part

Reply:
Since we don't have a real failure case for it for now, i intend to do this as a follow up

Reply:
Am bit occupied, since its very edge case i thought of deferring it to a later point. 
Feel free to raise a PR to gvisor-tap-vsock.

Reply:
Thanks, I'm fine. Please do it before landing this change.

Reply:
I don't think we need this upstream change as a hard dependency to this PR.

If time permits i will fix prior if not we will take as a follow up.

Reply:
I am not a maintainer, but I see this as completely unnecessary technical debt.